### DISCUSSION TITLE
New event type for Workflows: Merge

### DISCUSSION THREAD

**Discussion by TechTeam12** [community/community#25514] [Actions]
<p>I’ve got a lot of private packages I work with and I’ve built out a automated package manager publish that iterates the version number after publish then updates the master branch. This is for like a docker image then it’ll update the tag. This is workflow is a problem though because it enters an infinite loop. When my workflow which is looking for:</p>
<pre><code>name: pusher-step
on:
  push:
    branches:
      - master
</code></pre>
<p>As you can probably guess, after the publish happens we update the master branchs tags through a private action… Which as you can guess, causes another run of this workflow… rinse and repeat.</p>
<p>So, proposal:<br>
Create a new event type: merge<br>
This event type will run once on merge, but not push.</p>

**Comment by TechTeam12** [comment]
<p>With a sober mind, what I meant to state was:</p>
<p>I have private packages &amp; images hosted with GitHub. I’ve tried to build out the CD part of the CI/CD where after my PR is merged I trigger a publish and then update the tags &amp; refs to the main branch. This causes an infinite loop because the workflow is triggered on a push to the master branch…</p>
<p>Proposal:<br>
Create a new event type: <code>merge</code></p>
<p>This event is specific to a PR being merged, and will run once. It’s not associated to push.</p>

**Comment by chrispat** [comment]
<p>You could do this today with the something like</p>
<pre><code>on:
  pull_request:
    types: [closed]
</code></pre>
<p>And then put a conditional on the job to see if the pull request was merged by checking the event payload.</p>

**Comment by TechTeam12** [comment]
<p>So the action would contain a check to see if the PR was merged? How would the pull request merged part work I’m not sure I understand that? What event payload would I have to check?</p>
<p>The way my workflow runs: on closed run my private action. The action then publishes/builds image then updates the repo. I’m not sure what payload would be passed to the action that I could check. Is there a way to do something like that in the action.yml?</p>
<p>I originally thought about using the closed, but the issue with that is (I’m sure you’ve guessed it) that the PRs could be closed and not merged.</p>
<p>Also, question for the pondering mind:</p>
<p>Which is the better solution to implement?</p>
<ol>
<li>For every repository exists where one would want to do a CD solution; you would have to implement a two step process. A workflow that listens for pull request closed, followed by some further filter to inspect the payload of event to ensure it was merged.</li>
<li>As all the other successful CD providers have done create a <code>merged</code> event that can be listened to specifically</li>
</ol>
<p>as reference:<br>
Gitlab’s merged is called <a href="https://docs.gitlab.com/ee/ci/yaml/README.html#stages" rel="nofollow noopener">deploy</a></p>

**Comment by TechTeam12** [comment]
<p>Comment not found: Old comment deleted to merge into single comment.</p>

**Comment by TechTeam12** [comment, marked as answer]
<p>If anyone stumbles onto this, the solution is to do something like:</p>
<pre><code>name: Node-CI-PROD
on:
  pull_request:
    types: [closed]
    branches:
      - master

jobs:
  build:
    name: prd
    runs-on: ubuntu-16.04

    strategy:
      matrix:
        node-version: [10.x]

    steps:
      - name: use_node
        uses: actions/setup-node@master
        with:
          node-version: ${{ matrix.node-version }}
      - testing the merged
        if: github.event.pull_request.merged == true
        run: |
          node -e "console.log('HELLO WORLD!');"
</code></pre>
